diff -r -u cairo-1.4.12.org/src/cairo-win32-font.c cairo-1.4.12/src/cairo-win32-font.c
--- cairo-1.4.12.org/src/cairo-win32-font.c	Tue Nov 27 15:20:12 2007
+++ cairo-1.4.12/src/cairo-win32-font.c	Sat Dec  1 14:39:04 2007
@@ -1685,6 +1685,12 @@
     return 1. / ((cairo_win32_scaled_font_t *)scaled_font)->logical_scale;
 }
 
+void
+cairo_win32_scaled_font_get_metrics_factor2 (cairo_scaled_font_t *scaled_font, double *out)
+{
+    *out = cairo_win32_scaled_font_get_metrics_factor (scaled_font);
+}
+
 /**
  * cairo_win32_scaled_font_get_logical_to_device:
  * @scaled_font: a scaled font from the Win32 font backend
Only in cairo-1.4.12/src: cairo-win32-font.c.orig
diff -r -u cairo-1.4.12.org/src/cairo-win32.h cairo-1.4.12/src/cairo-win32.h
--- cairo-1.4.12.org/src/cairo-win32.h	Tue Nov 27 09:30:52 2007
+++ cairo-1.4.12/src/cairo-win32.h	Sat Dec  1 14:39:04 2007
@@ -81,6 +81,9 @@
 cairo_win32_scaled_font_get_metrics_factor (cairo_scaled_font_t *scaled_font);
 
 cairo_public void
+cairo_win32_scaled_font_get_metrics_factor2 (cairo_scaled_font_t *scaled_font, double *out);
+
+cairo_public void
 cairo_win32_scaled_font_get_logical_to_device (cairo_scaled_font_t *scaled_font,
 					       cairo_matrix_t *logical_to_device);
 
diff -r -u cairo-1.4.12.org/src/cairo.c cairo-1.4.12/src/cairo.c
--- cairo-1.4.12.org/src/cairo.c	Tue Nov 27 15:20:12 2007
+++ cairo-1.4.12/src/cairo.c	Sat Dec  1 14:39:04 2007
@@ -3083,6 +3083,12 @@
 }
 slim_hidden_def (cairo_get_tolerance);
 
+void
+cairo_get_tolerance2 (cairo_t *cr, double *out)
+{
+    *out = cairo_get_tolerance (cr);
+}
+
 /**
  * cairo_get_antialias:
  * @cr: a cairo context
@@ -3181,6 +3187,12 @@
     return _cairo_gstate_get_line_width (cr->gstate);
 }
 
+void
+cairo_get_line_width2 (cairo_t *cr, double *out)
+{
+    *out = cairo_get_line_width (cr);
+}
+
 /**
  * cairo_get_line_cap:
  * @cr: a cairo context
@@ -3221,6 +3233,12 @@
 cairo_get_miter_limit (cairo_t *cr)
 {
     return _cairo_gstate_get_miter_limit (cr->gstate);
+}
+
+void
+cairo_get_miter_limit2 (cairo_t *cr, double *out)
+{
+    *out = cairo_get_miter_limit (cr);
 }
 
 /**
diff -r -u cairo-1.4.12.org/src/cairo.h cairo-1.4.12/src/cairo.h
--- cairo-1.4.12.org/src/cairo.h	Tue Nov 27 15:20:12 2007
+++ cairo-1.4.12/src/cairo.h	Sat Dec  1 14:39:04 2007
@@ -1221,6 +1221,9 @@
 cairo_public double
 cairo_get_tolerance (cairo_t *cr);
 
+cairo_public void
+cairo_get_tolerance2 (cairo_t *cr, double *out);
+
 cairo_public cairo_antialias_t
 cairo_get_antialias (cairo_t *cr);
 
@@ -1233,6 +1236,9 @@
 cairo_public double
 cairo_get_line_width (cairo_t *cr);
 
+cairo_public void
+cairo_get_line_width2 (cairo_t *cr, double *out);
+
 cairo_public cairo_line_cap_t
 cairo_get_line_cap (cairo_t *cr);
 
@@ -1241,6 +1247,9 @@
 
 cairo_public double
 cairo_get_miter_limit (cairo_t *cr);
+
+cairo_public void
+cairo_get_miter_limit2 (cairo_t *cr, double *out);
 
 cairo_public int
 cairo_get_dash_count (cairo_t *cr);
